pkg/apis/sources/v1alpha1/source_types.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"knative.dev/pkg/apis"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+)
+
+// BaseSourceSpec provides spec components that all sources share. It does not include any
+// PodSpecable helpers.
+type BaseSourceSpec struct {
+	// Sink is a reference to an object that will resolve to a URI to send events to.
+	// +required
+	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+
+	// OutputFormat describes the CloudEvent output format the source should send events in.
+	// All formats are over HTTP.
+	// Defaults to binary.
+	// +optional
+	OutputFormat OutputFormatType `json:"outputFormat,omitempty"`
```

Comment:
BTW, we decided that this will be a feature of PodSpecableSources but not the sources spec.